### PR TITLE
migrate knative plugin models to resource metadata dynamic plugin

### PR DESCRIPTION
**Pull request by vikram-raj**
Fixes - https://issues.redhat.com/browse/ODC-6348

- Migrate model definition to dynamic extension

### CODE REVIEW

**Comment by rohitkrai03** [comment]
@vikram-raj Can you update this PR. The PR - https://github.com/openshift/console/pull/9013 that adds support for `ResourceMetadata` has been merged now.

**Review by rohitkrai03 — APPROVED** [review]
/approve

File: frontend/packages/knative-plugin/console-extensions.json [hunk 1]
```diff
@@ -1124,5 +1124,369 @@
...
+      "labelPlural": "%knative-plugin~CronJobSources%",
+      "abbr": "CJS"
+    }
+  },
+  {
+    "type": "console.model-metadata",
+    "properties": {
+      "model": {
+        "group": "sources.knative.dev",
+        "version": "v1beta2",
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/console-extensions.json:1245]: PingSource has the latest version to v1 so we can update

File: frontend/packages/knative-plugin/console-extensions.json [hunk 2]
```diff
@@ -1124,5 +1124,369 @@
     "flags": {
       "required": ["KNATIVE_SERVING_SERVICE", "KNATIVE_SERVING_REVISION", "KNATIVE_EVENTING"]
     }
+  },
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/console-extensions.json:1127]: @vikram-raj can we remove model definition and static plugins now as we have dynamic one's now

https://github.com/openshift/console/blob/master/frontend/packages/knative-plugin/src/plugin.tsx#L38-L43
https://github.com/openshift/console/blob/master/frontend/packages/knative-plugin/src/models.ts#L1-L482

File: frontend/packages/knative-plugin/console-extensions.json [hunk 3]
```diff
@@ -1124,5 +1124,369 @@
...
+  {
+    "type": "console.model-metadata",
+    "properties": {
+      "model": {
+        "group": "sources.knative.dev",
+        "version": "v1beta2",
+        "kind": "PingSource"
+      },
+      "color": "#005f60",
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/console-extensions.json:1248]:
```suggestion
      "color": "#7d1007",
```
color is different for eventing and serving resources, this color applies to all eventing resources

File: frontend/packages/knative-plugin/console-extensions.json [hunk 4]
```diff
@@ -1124,5 +1124,407 @@
...
+  },
+  {
+    "type": "console.model-metadata",
+    "properties": {
+      "model": {
+        "group": "operator.knative.dev"
+      },
+      "color": "#005f60"
+    }
+  },
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/console-extensions.json:1154]: at other occurrences it looks good to have color with the group for serving and eventing however here we can avoid as both `knativeServing` and `knativeEventing` has a same group so better to specify kind as well explicitly

File: frontend/packages/console-shared/src/utils/pod-ring-utils.ts [hunk 5]
```diff
@@ -11,7 +11,7 @@ import {
   SelfSubjectAccessReviewKind,
   HorizontalPodAutoscalerKind,
 } from '@console/internal/module/k8s';
-import { RevisionModel } from '@console/knative-plugin';
+import { REVISION_KIND } from '@console/knative-plugin';
```
Comment (invincibleJai) [inline frontend/packages/console-shared/src/utils/pod-ring-utils.ts:14]: we don't need to introduce const for those resources for which we still have static models so it'll like duplicate, we can let use the model for now and we can do clean up as another PR.

File: frontend/packages/knative-plugin/src/topology/__tests__/topology-knative-test-data.ts [hunk 6]
```diff
@@ -474,7 +477,7 @@ export const revisionObj: RevisionKind = {
     },
     ownerReferences: [
       {
-        apiVersion: `${ConfigurationModel.apiGroup}/${ConfigurationModel.apiVersion}`,
+        apiVersion: `${RouteModel.apiGroup}/${RouteModel.apiVersion}`,
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/topology/__tests__/topology-knative-test-data.ts:480]: why this change?

File: frontend/packages/knative-plugin/src/topology/__tests__/topology-knative-test-data.ts [hunk 7]
```diff
@@ -540,7 +543,7 @@ export const knativeRouteObj: RouteKind = {
     },
     ownerReferences: [
       {
-        apiVersion: `${ServiceModel.apiGroup}/${ServiceModel.apiVersion}`,
+        apiVersion: `${RouteModel.apiGroup}/${RouteModel.apiVersion}`,
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/topology/__tests__/topology-knative-test-data.ts:546]: do we need this?

File: frontend/packages/knative-plugin/src/components/add/import-types.ts [hunk 8]
```diff
@@ -1,34 +1,33 @@
...
-  InMemoryChannel: EventingIMCModel.kind,
-  KafkaChannel: EventingKafkaChannelModel.kind,
+  InMemoryChannel: EVENTING_IMC_KIND,
+  KafkaChannel: EVENTING_KAFKA_CHANNEL_KIND,
 };
 export const defaultChannels = {
-  InMemoryChannel: EventingIMCModel,
-  KafkaChannel: EventingKafkaChannelModel,
+  InMemoryChannel: EVENTING_IMC_KIND,
+  KafkaChannel: EVENTING_KAFKA_CHANNEL_KIND,
```
[the remark this answers is not in the document]
Comment (vikram-raj) [reply frontend/packages/knative-plugin/src/components/add/import-types.ts:31]: `Channels` is not being used. So, removing the `Channels`.

File: frontend/packages/knative-plugin/src/utils/get-knative-resources.ts [hunk 9]
```diff
@@ -207,7 +207,7 @@ export const knativeEventingResourcesBroker = (
       isList: true,
       kind: referenceForModel(EventingBrokerModel),
       namespace,
-      prop: EventingBrokerModel.plural,
+      prop: 'brokers',
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/utils/get-knative-resources.ts:210]: as we have models for these we can use that

**Comment by invincibleJai** [comment]
Thanks @vikram-raj , overall looks good but i see an issue with translation in details page

image

**Comment by vikram-raj** [comment]
> Thanks @vikram-raj , overall looks good but i see an issue with translation in details page
> 
> image

I fixed this issue here https://github.com/openshift/console/pull/10330/commits/0366b6bf28b703b70029e5910a0750025fb3bc0f PTAL. @invincibleJai @rohitkrai03

**Comment by rohitkrai03** [comment]
Did a re-review of the new changes, overall looks good. @invincibleJai You can add your `/lgtm` if it looks good to you too.

**Comment by invincibleJai** [comment]
@vikram-raj there something off with badges for serverless

image

**Comment by vikram-raj** [comment]
Propagate from epic:
/label docs-approved
/label px-approved

**Comment by karthikjeeyar** [comment]
Tested, works as expected
/label qe-approved
